cclient/serialization/tests/test_json.c

```diff
@@ -584,21 +589,23 @@ void test_serialize_attach_to_tangle(void) {
       trits_8019, NUM_TRITS_SERIALIZED_TRANSACTION,
       (const tryte_t*)TEST_2673_TRYTES_1, NUM_TRYTES_SERIALIZED_TRANSACTION,
       NUM_TRYTES_SERIALIZED_TRANSACTION));
-  hash_array_push(attach_req->trytes, trits_8019);
+  hash_array_push(trytes, trits_8019);
```

Comment:
I really think we should offer API functions like `attach_to_tangle_req_add_trytes` for this kind of operations. The end user is not supposed to know the underlying types and their functions.

Reply:
good point, see issue #805